homeassistant/core.py

```diff
@@ -981,28 +1011,30 @@ def call(self, domain, service, service_data=None, blocking=False):
             fut = asyncio.Future(loop=self._hass.loop)  # type: asyncio.Future
 
             @callback
-            def service_executed(event):
+            def service_executed(event: Event) -> None:
                 """Handle an executed service."""
                 if event.data[ATTR_SERVICE_CALL_ID] == call_id:
                     fut.set_result(True)
 
             unsub = self._hass.bus.async_listen(
                 EVENT_SERVICE_EXECUTED, service_executed)
 
-        self._hass.bus.async_fire(EVENT_CALL_SERVICE, event_data)
+            self._hass.bus.async_fire(EVENT_CALL_SERVICE, event_data)
 
-        if blocking:
             done, _ = await asyncio.wait(
                 [fut], loop=self._hass.loop, timeout=SERVICE_CALL_LIMIT)
             success = bool(done)
             unsub()
             return success
 
-    async def _event_to_service_call(self, event):
+        self._hass.bus.async_fire(EVENT_CALL_SERVICE, event_data)
```

Comment:
We fire twice?

Reply:
We fire once in any case. `if blocking:` ends with `return`.
With two separate `if blocking:` mypy wouldn't figure that it's the same condition.